src/pages/billing/Project.js

```diff
@@ -14,187 +14,176 @@ import * as StateHistory from 'src/libs/state-history'
+          okButton: h(ButtonPrimary, {
+            disabled: !selectedBilling || billingAccountName === selectedBilling,
+            onClick: async () => {
+              setShowBillingModal(false)
+              await updateBillingAccount(selectedBilling)
+            }
+          }, ['Ok'])
         }, [
-          h(NewUserCard, {
-            onClick: () => this.setState({ addingUser: true })
-          }),
-          div({ style: { flexGrow: 1 } },
-            _.map(member => {
-              return h(MemberCard, {
-                adminLabel: 'Owner',
-                userLabel: 'User',
-                member, adminCanEdit,
-                onEdit: () => this.setState({ editingUser: member }),
-                onDelete: () => this.setState({ deletingUser: member })
-              })
-            }, _.filter(({ email }) => Utils.textMatch(filter, email), projectUsers))
-          )
+          h(IdContainer, [id => h(Fragment, [
+            h(FormLabel, { htmlFor: id, required: true }, ['Select billing account']),
+            h(Select, {
+              id,
+              value: selectedBilling || billingAccountName,
+              isClearable: false,
+              options: _.map(({ displayName, accountName }) => ({ label: displayName, value: accountName }), billingAccounts),
+              onChange: ({ value: newAccountName }) => setSelectedBilling(newAccountName)
+            })
+          ])])
         ])
       ]),
-      addingUser && h(NewUserModal, {
-        adminLabel: 'Owner',
-        userLabel: 'User',
-        title: 'Add user to Billing Project',
-        footer: 'Warning: Adding any user to this project will mean they can incur costs to the billing associated with this project.',
-        addFunction: Ajax().Billing.project(projectName).addUser,
-        onDismiss: () => this.setState({ addingUser: false }),
-        onSuccess: () => this.refresh()
-      }),
-      editingUser && h(EditUserModal, {
-        adminLabel: 'Owner',
-        userLabel: 'User',
-        user: editingUser,
-        saveFunction: Ajax().Billing.project(projectName).changeUserRoles,
-        onDismiss: () => this.setState({ editingUser: false }),
-        onSuccess: () => this.refresh()
-      }),
-      !!deletingUser && h(DeleteUserModal, {
-        userEmail: deletingUser.email,
-        onDismiss: () => this.setState({ deletingUser: false }),
-        onSubmit: _.flow(
-          withErrorReporting('Error removing member from billing project'),
-          Utils.withBusyState(v => this.setState({ updating: v }))
-        )(async () => {
-          this.setState({ deletingUser: false })
-          await Ajax().Billing.project(projectName).removeUser(deletingUser.roles, deletingUser.email)
-          this.refresh()
-        })
-      }),
-      (refreshing || loadingBillingInfo || updatingAccount || updating) && spinnerOverlay
-    ])
-  }
-
-  componentDidUpdate(prevProps, prevState) {
-    const hasBillingScope = Auth.hasBillingScope()
```

Comment:
I'm confused about this logic and why it was there before. The new code is simpler but I can't figure out this excatly

Reply:
Yeah, previously it was sporadically writing this into state, now I'm just consulting Auth every time as the single source of truth rather than trying to duplicate the information.